### PR TITLE
enable sanitizers on CI

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -164,3 +135,49 @@ jobs:
...
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt-get install libgmp-dev libboost-python-dev libpython3.7-dev libboost-system-dev build-essential -y
+        cd src
+        make ${{ matrix.config }} -f Makefile.vdf-client
+
+    - name: Build vdf-client on Mac
+      if: startsWith(matrix.os, 'mac')
+      run: |
+        cd src
```
Comment (hoffmang9) [inline .github/workflows/build.yml:164]: `brew install boost`

**Comment by arvidn** [comment]
`2weso` appears to be timing out when built with instrumentation. Perhaps TSAN and ASAN builds need to run smaller problems

**Comment by arvidn** [comment]
I came across another issue in `1weso_test` and `2weso_test` where they would not exit with and error code in case the test failed. I fixed that in the top commit as well as allowed the CI to run fewer iterations for TSAN and ASAN builds

**Review by hoffmang9 — APPROVED** [review]
Looks good. Am going to run a couple vdf_benches on beast before merging

**Comment by hoffmang9** [comment]
I'm going to merge as this should fix non vdf_client use cases

**Pull request merged**